### PR TITLE
Separated and integrated checklist package into HCM application

### PR DESCRIPTION
**Pull request by tanishi-egov**

### CODE REVIEW

File: apps/health_campaign_field_worker_app/lib/router/app_router.dart [hunk 1]
```diff
@@ -190,7 +188,7 @@ class AppRouter extends _$AppRouter {
                     page: HouseholdAcknowledgementRoute.page,
                     path: 'household-acknowledgement',
                   ),
-                  AutoRoute(page: ChecklistViewRoute.page, path: 'view'),
+                  // AutoRoute(page: ChecklistViewRoute.page, path: 'view'),
```
Comment (naveenr-egov) [inline apps/health_campaign_field_worker_app/lib/router/app_router.dart:191]: Remove commented code

File: apps/health_campaign_field_worker_app/lib/widgets/showcase/showcase_button.dart [hunk 2]
```diff
@@ -73,18 +73,18 @@ class _ShowcaseButtonState extends LocalizedState<ShowcaseButton> {
...
-        );
-      case ChecklistBoundaryViewRoute.name:
-        return checklistDataShowcaseData.showcaseData.map(
-          (e) => e.showcaseKey,
-        );
-      case ChecklistPreviewRoute.name:
-        return checklistListShowcaseData.showcaseData.map(
-          (e) => e.showcaseKey,
-        );
+      // case ChecklistRoute.name:
```
Comment (naveenr-egov) [inline apps/health_campaign_field_worker_app/lib/widgets/showcase/showcase_button.dart:76]: Remove commented code

File: packages/checklist/lib/pages/checklist_view.dart [hunk 3]
```diff
@@ -1,21 +1,31 @@
 import 'dart:math';
 
-import 'package:digit_components/digit_components.dart';
+import 'package:auto_route/auto_route.dart';
+import 'package:checklist/checklist.dart';
+import 'package:checklist/utils/extensions/extensions.dart';
+import 'package:digit_components/blocs/location/location.dart';
```
Comment (naveenr-egov) [inline packages/checklist/lib/pages/checklist_view.dart:6]: Why is the digit_components relative import not used?

File: packages/checklist/lib/pages/checklist_view.dart [hunk 4]
```diff
@@ -294,8 +300,6 @@ class _ChecklistViewPageState extends LocalizedState<ChecklistViewPage> {
                           ),
                         );
                         if (shouldSubmit ?? false) {
-                          router.navigate(ChecklistRoute());
```
Comment (naveenr-egov) [inline packages/checklist/lib/pages/checklist_view.dart:297]: Why is this removed?

File: packages/checklist/lib/utils/constants.dart [hunk 5]
```diff
@@ -0,0 +1,11 @@
+class Constants {
+  static const String localizationApiPath = 'localization/messages/v1/_search';
```
Comment (naveenr-egov) [inline packages/checklist/lib/utils/constants.dart:2]: This should be taken from main app. This can be moved to singleton of the package.

File: packages/closed_household/pubspec.yaml [hunk 6]
```diff
@@ -43,6 +43,11 @@ dependencies:
   registration_delivery: ^1.0.3
 
 
+dependency_overrides:
```
Comment (naveenr-egov) [inline packages/closed_household/pubspec.yaml:46]: This can be removed since there is local override file

File: packages/registration_delivery/pubspec.lock [hunk 7]
```diff
@@ -427,19 +458,18 @@ packages:
...
   digit_scanner:
     dependency: "direct main"
     description:
       name: digit_scanner
-      sha256: "1f1fe3ca8f038a3c93cd52491cf72c0b0e45c88b14580243d47eac59672e67ea"
+      sha256: bc172291df5ffe720e3d471fbc5c542086b8efecbdb6e25ee737381cc47c8e05
       url: "https://pub.dev"
     source: hosted
-    version: "1.0.3"
+    version: "1.0.3-dev.1"
```
Comment (naveenr-egov) [inline packages/registration_delivery/pubspec.lock:472]: This was moved to stable please do pub get again.

**Review by naveenr-egov — CHANGES_REQUESTED** [review]

File: apps/health_campaign_field_worker_app/lib/router/app_router.dart (hunk 1 above)
Comment (tanishi-egov) [reply apps/health_campaign_field_worker_app/lib/router/app_router.dart:191]: > Why is this commented?

by mistake, removed now

File: packages/checklist/CHANGELOG.md [hunk 8]
```diff
@@ -0,0 +1,3 @@
+## 0.0.1
+
+* TODO: Describe initial release.
```
[the remark this answers is not in the document]
Comment (tanishi-egov) [reply packages/checklist/CHANGELOG.md:3]: > Describe the initial release

okay, done

File: packages/checklist/lib/blocs/service_definition.dart [hunk 9]
```diff
@@ -14,32 +14,33 @@ class ServiceDefinitionBloc
...
-    required this.serviceDefinitionDataRepository,
-  }) {
+      super.initialState, {
+        required this.serviceDefinitionDataRepository,
+      }) {
     on(_handleFetch);
     on(_handleSelect);
   }
 
+  //this function is called to fetch all the service definitions from the table for the selected project
```
[the remark this answers is not in the document]
Comment (tanishi-egov) [reply packages/checklist/lib/blocs/service_definition.dart:24]: > Sentence should start with Uppercase

done

File: packages/checklist/lib/checklist.dart [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+export 'data/repositories/local/service.dart';
+export 'data/repositories/local/service_definition.dart';
+export 'data/repositories/remote/service.dart';
+export 'data/repositories/remote/service_definition.dart';
+export 'data/repositories/remote/service_attributes.dart';
+export 'utils/utils.dart';
+export 'blocs/service.dart';
+export 'blocs/service_definition.dart';
+export 'blocs/app_localization.dart';
+// export 'utils/typedefs.dart';
```
[the remark this answers is not in the document]
Comment (tanishi-egov) [reply packages/checklist/lib/checklist.dart:23]: > Why is this commented?

We removed the comment now, (exporting typedef was giving conflicts in referal_reconcilition package)

File: packages/checklist/pubspec.yaml [hunk 11]
```diff
@@ -0,0 +1,84 @@
+name: checklist
+description: "A new Flutter package project."
```
[the remark this answers is not in the document]
Comment (tanishi-egov) [reply packages/checklist/pubspec.yaml:2]: > Add a description

done

File: packages/digit_data_model/lib/utils/typedefs.dart [hunk 12]
```diff
@@ -33,7 +33,7 @@ typedef ProductVariantDataRepository
 typedef BoundaryDataRepository
     = DataRepository<BoundaryModel, BoundarySearchModel>;
 
-typedef ServiceDataRepository
-    = DataRepository<ServiceModel, ServiceSearchModel>;
-typedef ServiceDefinitionDataRepository
-    = DataRepository<ServiceDefinitionModel, ServiceDefinitionSearchModel>;
+// typedef ServiceDataRepository
```
[the remark this answers is not in the document]
Comment (tanishi-egov) [reply packages/digit_data_model/lib/utils/typedefs.dart:36]: > Please remove commented code

okay

File: packages/digit_data_model/lib/data/repositories/local/base/service_attributes_base.dart
[the remark this answers is not in the document]
Comment (tanishi-egov) [reply packages/digit_data_model/lib/data/repositories/local/base/service_attributes_base.dart:1]: > Why is this not moved to checklist package?

base repositories were not being used that's why we didn't move it....we deleted it from data model package as well

File: packages/attendance_management/pubspec.yaml [hunk 13]
```diff
@@ -33,6 +33,11 @@ dependencies:
   location: ^5.0.0
   dio: ^5.1.2
 
+
+dependency_overrides:
```
[the remark this answers is not in the document]
Comment (tanishi-egov) [reply packages/attendance_management/pubspec.yaml:37]: > We are maintaining a local override file. This can be removed.
okayy

File: packages/closed_household/pubspec.yaml (hunk 6 above)
Comment (tanishi-egov) [reply packages/closed_household/pubspec.yaml:46]: > This can be removed since there is local override file

okay

File: packages/registration_delivery/pubspec.lock (hunk 7 above)
Comment (tanishi-egov) [reply packages/registration_delivery/pubspec.lock:472]: > This was moved to stable please do pub get again.

okayy

File: packages/checklist/lib/utils/constants.dart (hunk 5 above)
Comment (tanishi-egov) [reply packages/checklist/lib/utils/constants.dart:2]: > This should be taken from main app. This can be moved to singleton of the package.

This is not being used in the package, so we removed it from here

File: packages/checklist/lib/utils/extensions/extensions.dart [hunk 14]
```diff
@@ -0,0 +1,65 @@
...
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:intl/intl.dart';
+import '../utils.dart';
+
+// Importing the file 'context_utility.dart' with a 'part' directive.
+part 'context_utility.dart';
+
+// Extension to calculate age based on a DateTime object.
+extension DateAgeExtension on DateTime {
+  int get age {
```
[the remark this answers is not in the document]
Comment (tanishi-egov) [reply packages/checklist/lib/utils/extensions/extensions.dart:13]: > Please check if this is being used else delete this. Since age is not used in checklist.

We are deleting the extensions file as it is not being used anywhere

File: packages/checklist/lib/data/repositories/remote/service_definition.dart [hunk 15]
```diff
@@ -1,14 +1,14 @@
 // Generated using mason. Do not modify by hand
 
 import 'package:digit_data_model/data_model.dart';
-import '../../../models/entities/service_definition.dart';
+import '../../../models/entities/service_definition.dart'as svc_def;
```
[the remark this answers is not in the document]
Comment (tanishi-egov) [reply packages/checklist/lib/data/repositories/remote/service_definition.dart:4]: > Why is it imported as?

fixed it

File: packages/checklist/lib/models/entities/service_attributes.dart [hunk 16]
```diff
@@ -34,10 +31,10 @@ class ServiceAttributesModel extends EntityModel with ServiceAttributesModelMapp
   static const schemaName = 'ServiceAttributes';
 
   final String? attributeCode;
-  final dynamic? value;
+  final dynamic value;
```
[the remark this answers is not in the document]
Comment (tanishi-egov) [reply packages/checklist/lib/models/entities/service_attributes.dart:34]: > Why is this changed?

It's dynamic so it can be nullable....so no point in putting a '?' there.

File: .fvmrc
[the remark this answers is not in the document]
Comment (tanishi-egov) [reply .fvmrc:1]: > This should not pushed since the .fvm extensions are added in git-ignore. And we are using 3.16.5

okay

File: packages/checklist/lib/pages/checklist_view.dart (hunk 3 above)
Comment (tanishi-egov) [reply packages/checklist/lib/pages/checklist_view.dart:6]: > Why is the digit_components relative import not used?

Since it is inside a package, we are doing a package import and not relative

File: apps/health_campaign_field_worker_app/lib/widgets/showcase/showcase_button.dart (hunk 2 above)
Comment (tanishi-egov) [reply apps/health_campaign_field_worker_app/lib/widgets/showcase/showcase_button.dart:76]: > Why is it commented?

fixed it now

Comment (tanishi-egov) [reply apps/health_campaign_field_worker_app/lib/widgets/showcase/showcase_button.dart:76]: > Why is it commented?

done

File: packages/checklist/lib/pages/checklist_view.dart (hunk 4 above)
Comment (tanishi-egov) [reply packages/checklist/lib/pages/checklist_view.dart:297]: > Why is this removed?

fixed it

Comment (tanishi-egov) [reply packages/checklist/lib/pages/checklist_view.dart:297]: > Why is this removed?

resolved it

File: apps/health_campaign_field_worker_app/lib/router/app_router.dart [hunk 17]
```diff
@@ -190,7 +188,6 @@ class AppRouter extends _$AppRouter {
                     page: HouseholdAcknowledgementRoute.page,
                     path: 'household-acknowledgement',
                   ),
-                  AutoRoute(page: ChecklistViewRoute.page, path: 'view'),
```
Comment (naveenr-egov) [inline apps/health_campaign_field_worker_app/lib/router/app_router.dart:193]: Why is this removed? Survey form is been used in registration delivery as well right?

File: apps/health_campaign_field_worker_app/pubspec.yaml [hunk 18]
```diff
@@ -92,6 +99,7 @@ dev_dependencies:
   isar_generator: ^3.0.5
   flutter_launcher_icons: ^0.13.1
   hrk_flutter_test_batteries: ^0.8.0
+  change_app_package_name: ^1.3.0
```
Comment (naveenr-egov) [inline apps/health_campaign_field_worker_app/pubspec.yaml:102]: This should be removed since its not being used.

File: packages/registration_delivery/pubspec.lock
Comment (naveenr-egov) [inline packages/registration_delivery/pubspec.lock:1]: Whys is this deleted?

File: packages/survey_form/example/android/build.gradle [hunk 19]
```diff
@@ -0,0 +1,30 @@
+buildscript {
+    ext.kotlin_version = '1.9.0'
```
Comment (naveenr-egov) [inline packages/survey_form/example/android/build.gradle:2]: 1.8.22 is what we are using

File: packages/registration_delivery/pubspec.lock
Comment (tanishi-egov) [reply packages/registration_delivery/pubspec.lock:1]: > Why is this deleted? Please check other package lock files as well

it was giving issue during build in github so we deleted and re-generated it later

File: packages/survey_form/example/android/build.gradle (hunk 19 above)
Comment (tanishi-egov) [reply packages/survey_form/example/android/build.gradle:2]: > 1.8.22 is what we are using

okayy

File: apps/health_campaign_field_worker_app/pubspec.yaml (hunk 18 above)
Comment (tanishi-egov) [reply apps/health_campaign_field_worker_app/pubspec.yaml:102]: > This should be removed since its not being used.

removed

File: apps/health_campaign_field_worker_app/lib/router/app_router.dart (hunk 17 above)
Comment (Shashwat12-egov) [reply apps/health_campaign_field_worker_app/lib/router/app_router.dart:189]: In registration delivery that router path is not there that is why we removed from here as well.
